mod_form.php

```diff
@@ -51,6 +51,16 @@ public function definition() {
         $mform->addRule('name', null, 'required', null, 'client');
         $mform->addRule('name', get_string('maximumchars', 'core', 255), 'maxlength', 255, 'client');
 
+        // Add the option to display the content on the course page.
+        $mform->addElement('select', 'display', get_string('display', 'mod_poster'),
+            array(POSTER_DISPLAY_PAGE => get_string('displaypage', 'mod_poster'),
+                POSTER_DISPLAY_INLINE => get_string('displayinline', 'mod_poster')));
+        $mform->addHelpButton('display', 'display', 'mod_poster');
+        if (!$this->courseformat->has_view_page()) {
+            $mform->setConstant('display', POSTER_DISPLAY_PAGE);
+            $mform->hardFreeze('display');
+        }
+
```

Comment:
I would move the field elsewhere in the form. The original idea was to have the "Display name on view page" right below the "Name" field. I would move this setting to the very bottom of the General section probably.